### PR TITLE
Javascript carousel - Michael Chang

### PR DESCRIPTION
**Pull request by michael5522**

### CODE REVIEW

File: javascript-carousel/main.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    tabAll[1].classList.remove('active');
+    tabAll[2].classList.remove('active');
+    tabAll[3].classList.remove('active');
+    tabAll[4].classList.remove('active');
+    viewAll[0].classList.remove('hidden');
+    viewAll[1].classList.add('hidden');
+    viewAll[2].classList.add('hidden');
+    viewAll[3].classList.add('hidden');
+    viewAll[4].classList.add('hidden');
+  }
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js]: That's a _lot_ of repetition: try to combine these.

File: javascript-carousel/main.js [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+      viewAll[3].classList.add('hidden');
+      viewAll[4].classList.add('hidden');
+    }
+  });
+}
+var libraryCounter = 0;
+var library = [
+  1, 2, 3, 4, 5
+];
+setInterval(changeSlide, 3000);
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js]: The interval should restart every time the user interacts with the carousel controls.

File: javascript-carousel/main.js [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+      } else {
+        viewAll[j].classList.remove('hidden');
+      }
+    }
+  }
+});
+
+var backButtonAll = document.querySelectorAll('.previous');
+for (var i = 0; i < backButtonAll.length; i++) {
+  backButtonAll[i].addEventListener('click', function (event) {
```
Comment (TimelikeClosure) [inline javascript-carousel/main.js:41]: The forward and back buttons should only have one listener apiece, not 5.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Comment by michael5522** [comment]
https://user-images.githubusercontent.com/55678486/169216548-8c83eb5e-451d-40a9-bf30-e9b341f6d635.mov


https://user-images.githubusercontent.com/55678486/169216567-66cf6f3c-590f-4f28-aeaa-fa438aeae029.mov


https://user-images.githubusercontent.com/55678486/169216578-2ce80f98-95ff-42e7-8a98-44b278ab7990.mov

File: javascript-carousel/main.js (hunk 3 above)
Comment (TimelikeClosure) [reply javascript-carousel/main.js:41]: This still needs to be addressed.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**